### PR TITLE
feat: Add debug sharding logging utils

### PR DESCRIPTION
**Pull request by hlFu**
## Motivation

  Sharding mistakes in JAX models are easy to make and hard to spot — silently falling back from `P('tensor', ...)` to a replicated layout rarely throws, it just burns HBM and bandwidth. Today, debugging this means sprinkling ad-hoc `print(...)` calls in model code (e.g. duplicate `log_shardings` decorators were starting to appear in `layernorm.py` and `grok.py`),
  then ripping them out before merging.

  This PR introduces a single, env-gated debug utility for sharding inspection, and wires it into the Grok path.

  ## Modifications

  **New `python/sgl_jax/srt/utils/debug_utils.py`:**

  - `FrameworkLogLevel` (`IntEnum`): `ERROR=0`, `WARN=1`, `INFO=2`, `DEBUG=3`, `TRACE=4`.
  - `FRAMEWORK_LOG_LEVEL`: read once at import from `SGLANG_FRAMEWORK_LOG_LEVEL` (default `0`). `IntEnum` construction validates the value — unknown levels raise `ValueError` at startup rather than silently mis-gating.
  - `print_parameter_shardings(model)`: iterates `model.named_parameters()` and prints `name: shape=... sharding=...` for every parameter. No-op below `DEBUG`.
  - `log_shardings(name)`: decorator that prints each `__call__` input/output's shape + `NamedSharding` (reads `.aval.sharding` so it works inside `jit`). Returns the original function unwrapped below `DEBUG`, so there's zero runtime overhead at the default level.

  **Wiring:**

  - `python/sgl_jax/srt/model_loader/loader.py`: `JAXModelLoader.load_model` now calls `print_parameter_shardings(model)` after `load_weights` (replaces an unconditional inline loop).
  - `python/sgl_jax/srt/models/grok.py`: `Grok1MLP.__call__`, `Grok1MoE.__call__`, and `Grok1Attention.__call__` are decorated with `@log_shardings("GrokMLP" | "GrokMoE" | "GrokAttention")`.

  **Usage:**

  ```bash
  # default — silent
  python -m sgl_jax.launch_server ...

  # print parameter shardings at load time + activation shardings per forward
  SGLANG_FRAMEWORK_LOG_LEVEL=3 python -m sgl_jax.launch_server ...
```

**Sampled Outputs**
```
# Parameters
model.layers.1.self_attn.k_proj.weight: shape=(8192, 1024) sharding=NamedSharding(mesh=Mesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit)), spec=P(None, 'tensor'), memory_kind=device)
model.layers.1.self_attn.o_proj.weight: shape=(8192, 8192) sharding=NamedSharding(mesh=Mesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit)), spec=P('tensor', None), memory_kind=device)
model.layers.1.self_attn.q_proj.weight: shape=(8192, 8192) sharding=NamedSharding(mesh=Mesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit)), spec=P(None, 'tensor'), memory_kind=device)
model.layers.1.self_attn.v_proj.weight: shape=(8192, 1024) sharding=NamedSharding(mesh=Mesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit)), spec=P(None, 'tensor'), memory_kind=device)
model.layers.2.block_sparse_moe.experts.wi_0: shape=(8, 8192, 16384) sharding=NamedSharding(mesh=Mesh('expert': 1, 'tensor': 8, axis_types=(Explicit, Explicit)), spec=P('expert', 'tensor', None), memory_kind=device)
model.layers.2.block_sparse_moe.experts.wi_1: shape=(8, 8192, 16384) sharding=NamedSharding(mesh=Mesh('expert': 1, 'tensor': 8, axis_types=(Explicit, Explicit)), spec=P('expert', 'tensor', None), memory_kind=device)
model.layers.2.block_sparse_moe.experts.wo: shape=(8, 16384, 8192) sharding=NamedSharding(mesh=Mesh('expert': 1, 'tensor': 8, axis_types=(Explicit, Explicit)), spec=P('expert', None, 'tensor'), memory_kind=device)

# Activations
GrokMoE input[1]: (1, 8192) NamedSharding(mesh=AbstractMesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit), device_kind=TPU7x, num_cores=1), spec=P(None, None))
GrokMoE output[0]: (1, 8192) NamedSharding(mesh=AbstractMesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit), device_kind=TPU7x, num_cores=1), spec=P(None, None))
GrokMoE output[1]: (1, 2) NamedSharding(mesh=AbstractMesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit), device_kind=TPU7x, num_cores=1), spec=P(None, None))
dual_norm input[0]: (1, 8192) NamedSharding(mesh=AbstractMesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit), device_kind=TPU7x, num_cores=1), spec=P(None, None))
dual_norm input[1]: (1, 8192) NamedSharding(mesh=AbstractMesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit), device_kind=TPU7x, num_cores=1), spec=P(None, None))
dual_norm input[2]: (8192,) NamedSharding(mesh=AbstractMesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit), device_kind=TPU7x, num_cores=1), spec=P(None,))
dual_norm input[3]: (8192,) NamedSharding(mesh=AbstractMesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit), device_kind=TPU7x, num_cores=1), spec=P(None,))
norm input[0]: (1, 8192) NamedSharding(mesh=AbstractMesh('data': 1, 'tensor': 8, axis_types=(Explicit, Explicit), device_kind=TPU7x, num_cores=1), spec=P(None, None))
```

  Accuracy Tests

  No model-math changes — this PR only adds gated instrumentation. Forward numerics are identical between level=0 and level=3 (the decorator calls fn(*args, **kwargs) and returns its result verbatim).

  Benchmarking and Profiling

  Smoke-tested on Grok-2 with --num_hidden_layers=4, TP=8, TPU, launch-grok-4layers.sh + send_request_grok_no_profile.sh (bs=1, input=8192, output=4):

  ┌────────────────────────────┬──────────────────────────────────┬──────────────────────────────────────────┬─────────┬──────────────┬─────────────┐
  │ SGLANG_FRAMEWORK_LOG_LEVEL │ Parameter sharding lines at load │ Activation sharding lines during request │ Latency │  Input tput  │ Output tput │
  ├────────────────────────────┼──────────────────────────────────┼──────────────────────────────────────────┼─────────┼──────────────┼─────────────┤
  │ 0 (default)                │ 0                                │ 0                                        │ 21.44 s │ 402.59 tok/s │ 3.66 tok/s  │
  ├────────────────────────────┼──────────────────────────────────┼──────────────────────────────────────────┼─────────┼──────────────┼─────────────┤
  │ 3 (DEBUG)                  │ 43                               │ 229                                      │ 21.26 s │ 405.91 tok/s │ 3.71 tok/s  │
  └────────────────────────────┴──────────────────────────────────┴──────────────────────────────────────────┴─────────┴──────────────┴─────────────┘

  Latency deltas are within run-to-run noise — the decorator short-circuits to the original function at level=0, and at level=3 the prints happen during tracing (not per-step), so steady-state performance is unaffected.

  Checklist

  - Please use English, otherwise it will be closed.
  - The purpose of the PR, or link existing issues this PR will resolve.
  - The test plan, such as providing test command.
  - (Optional) The necessary documentation update.

### CODE REVIEW

File: python/sgl_jax/srt/model_loader/loader.py [hunk 1]
```diff
@@ -260,6 +261,9 @@ def _get_model(self, model_class: Any, model_config: ModelConfig) -> nnx.Module:
         else:
             logger.info("No quantization config found. Skipping quantization.")
         model.load_weights(model_config)
+
+        print_parameter_shardings(model)
```
Comment (alexnails) [inline python/sgl_jax/srt/model_loader/loader.py:265]: this can also be decorator / post action instead ?

**Review by alexnails — APPROVED** [review]
left small comment style wise, but only if you feel like changing to it. Approved

File: python/sgl_jax/srt/model_loader/loader.py (hunk 1 above)
Comment (hlFu) [reply python/sgl_jax/srt/model_loader/loader.py:265]: The decorator for logging input/output of module forward function is necessary because it needs to intercept the function call. I think here since we just log parameter variables inside models, directly printing it seems more straightforward.
